docs/en/reference/namingstrategy.rst

```diff
@@ -4,9 +4,13 @@ Implementing a NamingStrategy
 .. versionadded:: 2.3
 
 Using a naming strategy you can provide rules for generating database identifiers,
-column or table names when the column or table name is not given. This feature helps
+column or table names. This feature helps
 reduce the verbosity of the mapping document, eliminating repetitive noise (eg: ``TABLE_``).
 
+.. warning
+
+    The naming strategy is always overridden by annotations on an entity.
```

Comment:
Nice, quite straightforward. The only issue here is that entities can be mapped by XML files, hence my suggestion of saying "entity mapping".

Reply:
Might be perceived as a bit abstract for new users though, adding an example might help:

```suggestion
    The naming strategy is always overridden by entity mapping such as the `Table` annotation.
```

Reply:
I agree, I also suspect that people will be more likely to remember they've used XML mapping than an annotation copied from an example/another file and just updated.